nodejs/awsx/apigateway/apikey.ts

```diff
@@ -0,0 +1,149 @@
+     *  1 - an existing Usage Plan - the API Keys will be associated with the usage plan
+     *  2 - UsagePlanArgs with [usagePlan.apiStages] defined to define a new Usage Plan
+     *  3 - UsagePlanArgs with [apis] defined and [usagePlan.apiStages] NOT defined to define a new
+     *      Usage Plan
+     */
+    usagePlan?: aws.apigateway.UsagePlan | aws.apigateway.UsagePlanArgs;
+
+    /**
+     * The API keys you would like to create & associate with the usage plan. You can pass an array
+     *  that has a combination of:
+     *  1 - an existing APIKey
+     *  2 - ApiKeyArgs for a new APIKey
+     */
+    apiKeys?: Array<aws.apigateway.ApiKey | aws.apigateway.ApiKeyArgs>;
+}
+
+export interface AssociatedAPIKeys {
+    readonly usagePlan: aws.apigateway.UsagePlan;
+    readonly keys: Key[];
+}
+
+export interface Key {
+    apikey: aws.apigateway.ApiKey;
+    usagePlanKey: aws.apigateway.UsagePlanKey;
+}
+
+/**
+ * Helper function that allows you to quickly create API Keys and associate them with an API.
+ *
+ * @param name The _unique_ name of the resource.
+ * @param args The arguments to use to populate this resource's properties.
+ */
+export function createAssociatedAPIKeys(name: string, args: APIKeyArgs): AssociatedAPIKeys {
+    const usagePlan = getUsagePlan(name, args);
+    const keys = getKeys(name, args, usagePlan);
+
+    return { usagePlan, keys };
+}
+
+/** @internal */
+function getUsagePlan(name: string, args: APIKeyArgs): aws.apigateway.UsagePlan {
+    if (args.usagePlan && pulumi.CustomResource.isInstance(args.usagePlan)) {
+        if (args.apis) {
+            throw new Error("cannot define both [args.apis] and an existing usagePlan [args.usagePlan]");
+        }
+        return args.usagePlan;
+    } else {
+        let usagePlanArgs: aws.apigateway.UsagePlanArgs | undefined = args.usagePlan;
+
+        if (args.apis) {
+            const stages = [];
+
+            for (const a of args.apis) {
+                stages.push({
+                    apiId: a.restAPI.id,
+                    stage: a.stage.stageName,
+                });
+            }
+
+            if (args.usagePlan) {
+                if (args.usagePlan.apiStages) {
+                    throw new Error("cannot define both [args.apis] and [args.usagePlan.apiStages]");
+                }
+
+                usagePlanArgs = {
+                    apiStages: stages,
+                    description: args.usagePlan.description,
+                    name: args.usagePlan.name,
+                    productCode: args.usagePlan.productCode,
+                    quotaSettings: args.usagePlan.quotaSettings,
+                    throttleSettings: args.usagePlan.throttleSettings,
+                };
+            } else {
+                usagePlanArgs = {
+                    apiStages: stages,
+                };
+            }
+        }
+        return new aws.apigateway.UsagePlan(name, usagePlanArgs);
```

Comment:
i'd like something in the docs to mention that you can omit both args.usagePlan  and args.apis and you'll still get a UsagePLan created on your behalf with all default values.